### PR TITLE
[BUGFIX] Fix for icons in list view

### PR DESCRIPTION
**Pull request by jmverges**
Closes: #857

### CODE REVIEW

**Comment by xf-** [comment]
I know its easiest way, but this is in usability a button with a action. no reaction means it is broken.
I will try to find something else.

**Comment by xf-** [comment]
hi,
```diff
diff --git a/Classes/Hooks/ContentIconHookSubscriber.php b/Classes/Hooks/ContentIconHookSubscriber.php
index 5b506e2..4ba81dd 100644
--- a/Classes/Hooks/ContentIconHookSubscriber.php
+++ b/Classes/Hooks/ContentIconHookSubscriber.php
@@ -24,8 +24,8 @@ use TYPO3\CMS\Recordlist\RecordList\DatabaseRecordList;
 class ContentIconHookSubscriber {
 
 	protected $templates = array(
-		'iconWrapper' => '<span class="btn btn-default"><span class="t3-icon t3-icon-empty t3-icon-empty-empty"
-								style="vertical-align: top;">%s</span></span>'
+		'iconWrapper' => '</div><style>.t3-js-clickmenutrigger {z-index: 2;position: relative}.t3-js-clickmenutrigger .t3-icon-pagetree-root{opacity: 0}</style><span class="t3-icon t3-icon-empty t3-icon-empty-empty"
+								style="margin-top: 7px;position: absolute;left: 12px;z-index: 1">%s</span><div style="display: none">'
 	);
 
 	/**
@@ -98,8 +98,10 @@ class ContentIconHookSubscriber {
 								$label = trim($form->getLabel());
 								$icon = '<img width="16" height="16" src="' . $icon . '" alt="' . $label . '"
 									title="' . $label . '" class="" />';
-								$icon = sprintf($this->templates['iconWrapper'], $icon);
+							} else {
+								$icon = '<span class="t3-icon t3-icon-apps t3-icon-apps-pagetree t3-icon-pagetree-root"> </span>';
 							}
+							$icon = sprintf($this->templates['iconWrapper'], $icon);
 						}
 					}
 				}
```

I know @NamelessCoder you hate it, but it looks good really good in FE. Also click ability is present and fallback to T3 icon.


1 Fluidcontent with hover example and fallback
2 row system element
3,4 Fluidcontent with icon

**Comment by jmverges** [comment]
+1

**Comment by xf-** [comment]
maybe we can add the css one time and also inline wouldn't needed anymore.
Update your PR and we have a nice FE :)
Maybe the same hacky thing in page and would reduce the confusion.

File: Classes/Backend/Preview.php [hunk 1]
```diff
@@ -135,6 +135,7 @@ protected function attachAssets() {
 			$doc->backPath = $GLOBALS['BACK_PATH'];
 
 			$doc->getPageRenderer()->addCssFile($doc->backPath . ExtensionManagementUtility::extRelPath('flux') . 'Resources/Public/css/grid.css');
+			$doc->getPageRenderer()->addCssFile($doc->backPath . ExtensionManagementUtility::extRelPath('flux') . 'Resources/Public/css/icon.css');
```
Comment (xf-) [inline Classes/Backend/Preview.php]: We don't need that line, was me mistake. The cache in the modules are really persistent.

File: Classes/Hooks/ContentIconHookSubscriber.php [hunk 2]
```diff
@@ -109,9 +103,12 @@ public function addSubIcon(array $parameters, $caller = NULL) {
 								$label = trim($form->getLabel());
 								$icon = '<img width="16" height="16" src="' . $icon . '" alt="' . $label . '"
 									title="' . $label . '" class="" />';
-								$icon = '<span class="t3-icon t3-icon-empty t3-icon-empty-empty"
-									style="float: left; vertical-align: bottom; margin-top: 2px;">' . $icon . '</span>';
+							} else {
+								$icon = '<span class="t3-icon t3-icon-apps t3-icon-apps-pagetree t3-icon-pagetree-root"> </span>';
 							}
+							$icon = sprintf($this->templates['iconWrapper'], $icon);
+							$this->attachAssets();
```
Comment (xf-) [inline Classes/Hooks/ContentIconHookSubscriber.php]: Because of persisent cache we should add it always to avoid problems.

**Pull request merged**